types/types.go

```diff
@@ -824,6 +824,13 @@ func ValidateDataLocalityAndReplicaCount(mode longhorn.DataLocality, count int)
 	return nil
 }
 
+func ValidateDataLocalityAndAccessMode(locality longhorn.DataLocality, mode longhorn.AccessMode) error {
+	if mode == longhorn.AccessModeReadWriteMany && locality == longhorn.DataLocalityStrictLocal {
```

Comment:
Also need to check `v.spec.migratable`. We only check the real RWX volume whose volume.spec.migratable is `false`.
 
```suggestion
	if mode == longhorn.AccessModeReadWriteMany && migratable == false, locality == longhorn.DataLocalityStrictLocal {
```

Reply:
Good catch! cc @PhanLe1010